cpp/src/io/parquet/page_enc.cu

```diff
@@ -1651,15 +1863,24 @@ static __device__ int32_t calculate_boundary_order(const statistics_chunk* s,
   return BoundaryOrder::UNORDERED;
 }
 
+// align ptr to an 8-byte boundary. address returned will be <= ptr.
+constexpr __device__ void* align8(void* ptr)
```

Comment:
Was blissfully unaware of `round_down_safe`.  Will give it a look, thanks.  I don't think this needs to be high performance since it's only called once per column chunk.

Reply:
`round_down_safe` is a host function :cry:

Reply:
Boo. `round_up_unsafe` can be used on device because it is constexpr. We can either add `constexpr` to `round_down_safe` or we can punt for now.

Reply:
There is a lot of pointer alignment all over the place (that's where I stole this code from), but it's always just inline with the code.  `round_down_safe` also suffers from expecting the value to round and the modulus to be the same type, so there'd be a lot of casting going on to make it work with a pointer.  Maybe change this to `template<typename P> P* align_pointer_to(P* ptr_to_round, uint8_t modulus)` and move it to utilities?

Reply:
We have an issue for it now. I would move on and leave it for now.